front-end/src/views/user/Login.vue

```diff
@@ -0,0 +1,232 @@
+      loginBtn: false,
+      // login type: 0 email, 1 username, 2 telephone
+      loginType: 0,
+      isLoginError: false,
+      form: this.$form.createForm(this),
+      state: {
+        time: 60,
+        loginBtn: false,
+        // login type: 0 email, 1 username, 2 telephone
+        loginType: 0,
+        smsSendBtn: false
+      },
+      needRememberMe: false,
+      validateRules: {
+        usernameRule: [{ required: true, message: this.$t('user.userName.required') }, { validator: this.handleUsernameOrEmail }],
+        passwordRule: [{ required: true, message: this.$t('user.password.required') }],
+        mobileRule: [{ required: true, message: this.$t('user.login.mobile.placeholder'), pattern: /^1[34578]\d{9}$/ }],
+        verificationCodeRule: [{ required: true, message: this.$t('user.verification-code.required') }]
+      }
+    }
+  },
+  created () {
+    const needRememberMe = storage.get(REMEMBER_ME)
+    if (needRememberMe === 'true') {
+      this.needRememberMe = true
+    }
+  },
+  methods: {
+    ...mapActions(['Login', 'Logout']),
+    handleUsernameOrEmail (rule, value, callback) {
+      const { state } = this
+      const regex = /^([a-zA-Z0-9_-])+@([a-zA-Z0-9_-])+((\.[a-zA-Z0-9_-]{2,3}){1,2})$/
+      if (regex.test(value)) {
+        state.loginType = 0
+      } else {
+        state.loginType = 1
+      }
+      callback()
+    },
+    handleSubmit (e) {
+      e.preventDefault()
+      const { form: { validateFields }, state, Login } = this
+
+      state.loginBtn = true
+
+      validateFields(['username', 'password'], { force: true }, (err, values) => {
+        if (!err) {
+          const loginParams = { ...values, password: md5(values.password) }
+
+          if (this.needRememberMe) {
+            storage.set(REMEMBER_ME, 'true')
+            storage.set(LOGIN_INFO, JSON.stringify(loginParams))
+          } else {
+            storage.set(REMEMBER_ME, 'false')
+          }
+
+          Login(loginParams)
+            .then(res => this.loginSuccess(res))
+            .catch(err => this.requestFailed(err))
+            .finally(() => {
+              state.loginBtn = false
+            })
+        } else {
+          setTimeout(() => {
+            state.loginBtn = false
+          }, 600)
+        }
+      })
+    },
+    getCaptcha (e) {
+      e.preventDefault()
+      const { form: { validateFields }, state } = this
+
+      validateFields(['mobile'], { force: true }, (err, values) => {
+        if (!err) {
+          state.smsSendBtn = true
+
+          const interval = window.setInterval(() => {
+            if (state.time-- <= 0) {
+              state.time = 60
+              state.smsSendBtn = false
+              window.clearInterval(interval)
+            }
+          }, 1000)
+
+          const hide = this.$message.loading('验证码发送中..', 0)
+          getSmsCaptcha({ mobile: values.mobile }).then(res => {
+            setTimeout(hide, 2500)
+            this.$notification['success']({
```

Comment:
Codacy has a fix for the issue: [["success"] is better written in dot notation.](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834763)
```suggestion
            this.$notification.success({
```